ppafm/ml/Generator.py

```diff
@@ -389,29 +395,33 @@ def __init__(
         self.df_steps = self.afmulator.df_steps
         self.z_size = self.scan_dim[2] - self.df_steps + 1
 
-    def _prepareBuffers(self, rhos=None, rho_deltas=None):
+    def _prepare_tip_buffers(self, rhos=None, rho_deltas=None, density_cutoff=None):
         if rhos is None:
-            self.rhos = self.ffts = [(None, None)] * len(self.iZPPs)
+            self.rhos_tip = self.rhos_tip_delta = self.ffts_tip = self.ffts_tip_delta = [None] * len(self.iZPPs)
```

Comment:
There is a big difference now, from `[(None, None)] * len(self.iZPPs)` to `[None] * len(self.iZPPs)` Are you exchanging an original double list-like objects with 2 single-list object?.
I am also afraid by the connection of many variables together.

Reply:
The change in the list format is because I wanted to separate the two different tip arrays into separate variables. This if-branch is the case when the tip density array is not set and point charges are used instead, so it just sets everything to `None`.